### PR TITLE
Configure analytics for Relay product pages #13322

### PR DESCRIPTION
**Pull request by stephaniehobson**
## One-line summary

Configure analytics for Relay product pages #13322

## Significant changes and points to review

- add analytics to product settings
- add CTAs to buttons
- add js for tracking begin_checkout and attribution

## Issue / Bugzilla link

#13322

## Testing

https://www-demo5.allizom.org/en-US/products/relay/

In [GTM](https://tagmanager.google.com/#/container/accounts/52671733/containers/1075029/workspaces/1000220):

- Click "preview" to open the tag assistant website.
- Add the domain "https://www-demo5.allizom.org/en-US/products/relay/" and connect to it
- Disable add-ons and toggle off DNT, etc until tag assistant indicates it has a connection
- Click things on the Relay site that should have tracking on them
- Return to the tag assistant window and look for the CTA Clicks and Begin Checkout events

Check:

- [x] All buttons fire a cta click event for GA3
- [x] Waitlist signups fire a `newsletter subscription` event for GA3
- [x] begin_checkout event is fired for GA4 for any _paid_ subscription referral to FxA

### CODE REVIEW

File: bedrock/products/templates/products/relay/includes/macros.html [hunk 1]
```diff
@@ -13,57 +13,59 @@
...
+    country_code=country_code,
+    lang=LANG,
+    optional_parameters={
+      'utm_campaign': _utm_campaign
+    },
+    optional_attributes={
+      'data-cta-type': 'fxa-relay',
+      'data-cta-text': 'Sign Up - {product} - monthly'.format(product=product),
+      'data-cta-position': 'matrix - {position}'.format(position=position)
+    })
```
Comment (nathan-barrett) [inline bedrock/products/templates/products/relay/includes/macros.html:49]: nice use of `format` here 🍏

File: bedrock/products/templates/products/relay/includes/matrix.html [hunk 2]
```diff
@@ -85,7 +85,15 @@
...
-            {{ relay_fxa_button(entrypoint='mozilla.org-relay-landing', button_text=ftl('plan-matrix-get-relay-cta'), class_name="mzp-c-button mzp-t-product mzp-t-lg") }}
+            {{ relay_fxa_button(
+                entrypoint='mozilla.org-relay-landing',
+                button_text=ftl('plan-matrix-get-relay-cta'),
+                class_name="mzp-c-button mzp-t-product mzp-t-lg",
+                optional_attributes={
+                  'data-cta-type': 'cta: fxa-relay',
+                  'data-cta-text': 'Sign Up - relay-free',
+                  'data-cta-position': 'matrix - table'}
+              ) }}
```
Comment (nathan-barrett) [inline bedrock/products/templates/products/relay/includes/matrix.html:96]: **nit**: spacing on the closing bracket for `optional_attributes`
```suggestion
                optional_attributes={
                  'data-cta-type': 'cta: fxa-relay',
                  'data-cta-text': 'Sign Up - relay-free',
                  'data-cta-position': 'matrix - table'
                 }
              ) }}
```

File: bedrock/products/templates/products/relay/includes/matrix.html [hunk 3]
```diff
@@ -150,7 +158,15 @@ <h3 class="mzp-u-title-xs">{{ ftl('plan-matrix-heading-plan-free') }}</h3>
...
             <em class="c-matrix-footer-free">{{ ftl('plan-matrix-price-free') }}</em>
-            {{ relay_fxa_button(entrypoint='mozilla.org-relay-landing', button_text=ftl('plan-matrix-get-relay-cta'), class_name="mzp-c-button mzp-t-product mzp-t-lg") }}
+            {{ relay_fxa_button(
+              entrypoint='mozilla.org-relay-landing',
+              button_text=ftl('plan-matrix-get-relay-cta'),
+              class_name="mzp-c-button mzp-t-product mzp-t-lg",
+              optional_attributes={
+                'data-cta-type': 'cta: fxa-relay',
+                'data-cta-text': 'Sign Up - relay-free',
+                'data-cta-position': 'matrix - mobile'}
```
Comment (nathan-barrett) [inline bedrock/products/templates/products/relay/includes/matrix.html:168]: Same as above, but its just me being particular 😅

File: bedrock/settings/base.py [hunk 4]
```diff
@@ -1796,35 +1796,83 @@ def before_send(event, hint):
...
             "yearly": {
                 "id": "price_1LYCqyJNcmPzuWtR3Um5qDPu",
                 "price": 1.00,
                 "currency": "CHF",
+                "analytics": {
+                    "brand": "relay",
+                    "plan": "relay-email",
+                    "currency": "CHF",
+                    "discount": "6.00",
+                    "price": "12.00",
```
Comment (nathan-barrett) [inline bedrock/settings/base.py:1817]: **question**: Curious why the price is `12.00` here but above in the same object the price is `1.00`?

**Review by nathan-barrett — APPROVED** [review]
GTM dashboard looks good, and just had a few non-blocking comments r+!

File: bedrock/settings/base.py (hunk 4 above)
Comment (stephaniehobson) [reply bedrock/settings/base.py:1817]: Yearly vs monthly pricing (if I understand the question properly). At checkout for monthly pricing they pay 2.00 at checkout for yearly they pay 12.00.

**Pull request merged**